raiden/api/python.py

```diff
@@ -973,6 +976,7 @@ def transfer_async(
             identifier=identifier,
             secret=secret,
             secrethash=secrethash,
+            lock_timeout=lock_timeout,
```

Comment:
This should be validated prior sending it through the state machine.
```
lock_timeout_valid = (
    lock_timeout <= channel_state.settle_timeout and
    lock_timeout > channel_state.reveal_timeout
)
```
Copied from `is_channel_usable`.

We should make sure the value is valid first.

Also, please add a regression test for this to make sure invalid values are rejected.

Reply:
How can this be validated here without a channel in hand?
also, how do you define an invalid value?  

Note that there are test cases that covers invalid values (values for which there are no route to target).

Reply:
> How can this be validated here without a channel in hand?

You could use 
```
    channel_state = views.get_channelstate_for(
            chain_state=chain_state,
            token_network_registry_address=registry_address,
            token_address=token_address,
            partner_address=partner_address,
        )
```

> also, how do you define an invalid value?

You can see that `transfer_async` function has a set of validations raising `InvalidXError`. So you could introduce this new exception `InvalidLockTimeoutError` and raise it if the value is invalid.

> Note that there are test cases that covers invalid values (values for which there are no route to target).

Right, so we can use those existing tests to add one more check for the invalid lock timeout.